lib/CapabilityInvocation.js

```diff
@@ -0,0 +1,90 @@
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * Redistributions of source code must retain the above copyright notice,
+ * this list of conditions and the following disclaimer.
+ *
+ * Redistributions in binary form must reproduce the above copyright
+ * notice, this list of conditions and the following disclaimer in the
+ * documentation and/or other materials provided with the distribution.
+ *
+ * Neither the name of the Digital Bazaar, Inc. nor the names of its
+ * contributors may be used to endorse or promote products derived from
+ * this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS
+ * IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED
+ * TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A
+ * PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED
+ * TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+ * PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
+ * LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+ * NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+'use strict';
+
+const jsigs = require('jsonld-signatures');
+const utils = require('./utils');
+const vocab = require('./vocab');
+
+const {ProofPurpose} = jsigs;
+
+module.exports = class CapabilityInvocation extends ProofPurpose {
+  constructor(injector) {
+    super(injector);
+    this.documentLoader = injector.use('jsonld').documentLoader;
+    this.uri = vocab.capabilityInvocationUri;
+  }
+
+  // Arguments:
+  //  - document: An already-expanded version of the invocation document
+  //    this proof is attached to
+  //  - proof: An already-expanded version of the proof we are checking
+  //
+  // proofPurposeOptions keywords:
+  //  - expectedTarget: the target we expect this capability to apply to (URI).
+  //  - caveatVerifiers: a hashmap of caveat URIs to procedures which
+  //    verify those caveats.  The caveat-verifying procedures should take
+  //    two arguments: the caveat being verified, and the expanded invocation
+  //    document.
+  async verify(document, proof, proofPurposeOptions) {
+    const {capabilityUri, creatorUri} = vocab;
+
+    // Retrieve the JSON-LD document associated with the capability ID
+    const {document: capability} = await this.documentLoader(
+      utils.getOneOrDie(proof[capabilityUri]));
+
+    // Prepare to check that the creator matches the parentCapability's invoker
+    const creator = utils.getOneOrDie(proof[creatorUri]);
+    const parentInvoker = await utils.getParentInvoker({
```

Comment:
Ah!  I just realized there's a bug with this which I didn't think about when Ganesh and I talked about grabbing authority from the parent yesterday... it's easy to fix... while it's true in delegation we want to check the authorized invoker from the parent, we don't do that in invocation, we grab it from the immediate document.  Consider the `{C {B {A}}}` delegation chain that @gannan08 and I discussed on IRC on monday... in that case, we wouldn't want to check whether this invocation was the one authorized in B, they'd be C.  So while in delegation we check the parent, in invocation we check the immediate capability document.

Reply:
Fix https://github.com/digitalbazaar/ocapld.js/pull/1/commits/d6ce7a1afb54c4ca679ec5b104ed6664ff82b5d9
@cwebber Is this a sufficient fix?

Reply:
Seems good to me!